x-pack/plugins/observability_solution/metrics_data_access/server/lib/metrics/index.ts

```diff
@@ -39,6 +39,14 @@ export const query = async (
     },
   };
   const hasGroupBy = Array.isArray(options.groupBy) && options.groupBy.length > 0;
+  const groupInstanceFilter =
+    options.groupInstance?.reduce<Array<Record<string, any>>>((acc, group, index) => {
+      const key = options.groupBy?.[index];
+      if (typeof key === 'string' && group) {
```

Comment:
why do we need to check type of key?

Reply:
Not sure if this situation ever happens, but I checked it because the key can be null or undefined according to type:
![image](https://github.com/elastic/kibana/assets/12370520/17410d66-f865-40d1-b398-14ec1b817e2b)

Reply:
> I checked it because the key can be null or undefined according to type

Then we don't need the type check because we know it's an optional string:

```suggestion
      if (key && group) {
```

Reply:
Fixed in [5ff89e3](https://github.com/elastic/kibana/pull/181689/commits/5ff89e38d702807e6f1171b0e546047a87d15b40)